### PR TITLE
RFC: `FlightInfoBuilder` for helping create FlightInfo structures

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Sketch of a solution for https://github.com/apache/arrow-rs/issues/4281

# Rationale for this change
 
It is a PITA to create `FlightInfo` as described on https://github.com/apache/arrow-rs/issues/4281

# What changes are included in this PR?

This PR builds on @roeap 's https://github.com/apache/arrow-rs/pull/4266 and sketches out a:
1.  `FlightInfoBuilder` in c9ebeb7b495c52827afc26432fa377af23bc1c2a 
2.  Demonstrates how it could be used in 623f3bbc904c0db0c30aa36f804ab38a7c59dc89

# Are there any user-facing changes?
It would be a new API

I think we could use this new builder to clean up the code a bit in IOx -- https://github.com/influxdata/influxdb_iox/blob/cdd519424dd102cac5162899fcb0fde21103a1dc/service_grpc_flight/src/lib.rs#L657-L680

If this seems like a reasonable idea I'll polish up the code and get a PR ready

### CODE REVIEW

File: arrow-flight/src/lib.rs [hunk 1]
```diff
@@ -444,6 +461,112 @@ impl FlightInfo {
...
+    /// Create a new `FlightInfoBuilder`
+    pub fn new() -> Self {
+        Self {
+            inner: FlightInfo {
+                schema: Bytes::new(),
+                flight_descriptor: None,
+                endpoint: vec![],
+                ordered: false,
+
+                // Flight says "Set these to -1 if unknown."
```
Comment (avantgardnerio) [inline arrow-flight/src/lib.rs:486]: Oh, cool, I did not know that!

**Review by avantgardnerio — APPROVED** [review]
Cleaner + less unimplemented, what's not to love?

File: arrow-flight/src/lib.rs (hunk 1 above)
Comment (alamb) [reply arrow-flight/src/lib.rs:486]: yeah, this is a good example of where the protobuf defaults (0) is different than what the spec says

**Comment by alamb** [comment]
Superceded by https://github.com/apache/arrow-rs/pull/4294

**Pull request closed without merging**